lib/MySQL_HostGroups_Manager.cpp

```diff
@@ -8253,6 +8254,119 @@ void MySQL_HostGroups_Manager::HostGroup_Server_Mapping::remove_HGM(MySrvC* srv)
 	srv->ConnectionsFree->drop_all_connections();
 }
 
+/**
+* @brief Updates replication hostgroups by adding autodiscovered mysql servers.
+* @details Adds each server from 'new_servers' to the 'runtime_mysql_servers' table.
+* We then rebuild the 'mysql_servers' table as well as the internal 'hostname_hostgroup_mapping'.
+* @param new_servers A vector of tuples where each tuple contains the values needed to add each new server.
+*
+* @return Returns EXIT_FAILURE code on failure and EXIT_SUCCESS code on success.
+*/
+int MySQL_HostGroups_Manager::add_discovered_servers_to_mysql_servers_and_replication_hostgroups(vector<tuple<string, int, int>> new_servers) {
+	int exit_code = EXIT_SUCCESS;
+	bool added_new_server = false;
+	wrlock();
+
+	try {
+		for (tuple<string, int, int> s : new_servers) {
+			string host = std::get<0>(s);
+			uint16_t port = std::get<1>(s);
+			long int hostgroup_id = std::get<2>(s);
+
+			// Add the discovered server with default values
+			MySrvC* mysrvc = new MySrvC(
+				const_cast<char*>(host.c_str()), port, 0, 1, MYSQL_SERVER_STATUS_ONLINE, 0, -1, 0, -1, 0, const_cast<char*>("Discovered endpoint")
+			);
+			add(mysrvc, hostgroup_id);
+
+			proxy_info(
+				"Adding new discovered server %s:%d with: hostgroup=%d, weight=%ld, max_connections=%ld, use_ssl=%d\n",
+				host.c_str(), port, hostgroup_id, mysrvc->weight, mysrvc->max_connections, mysrvc->use_ssl
+			);
+
+			added_new_server = true;
+		}
+
+		if (added_new_server) {
+			purge_mysql_servers_table();
+			mydb->execute("DELETE FROM mysql_servers");
+			proxy_debug(PROXY_DEBUG_MYSQL_CONNPOOL, 4, "DELETE FROM mysql_servers\n");
+			generate_mysql_servers_table();
```

Comment:
This block, starting at previous `if` is missing the update on the global `checksums` after this line. This section can be taken as an example [update_group_replication_add_autodiscovered](https://github.com/sysown/proxysql/blob/v2.x/lib/MySQL_HostGroups_Manager.cpp#L5893). Here we can see that the following part is missing:

```
		// Update the global checksums after 'mysql_servers' regeneration
		{
			unique_ptr<SQLite3_result> resultset { get_admin_runtime_mysql_servers(mydb) };
			string mysrvs_checksum { get_checksum_from_hash(resultset ? resultset->raw_checksum() : 0) };
			save_runtime_mysql_servers(resultset.release());
			proxy_info("Checksum for table %s is %s\n", "mysql_servers", mysrvs_checksum.c_str());

			pthread_mutex_lock(&GloVars.checksum_mutex);
			update_glovars_mysql_servers_checksum(mysrvs_checksum);
			pthread_mutex_unlock(&GloVars.checksum_mutex);
		}
```

Checksum re-computation is mandatory in this case after re-generating `mysql_servers_table`. But in this case there are a couple of extra operations that are required:

```
			// Update the global checksums after 'mysql_servers' regeneration
			{
				unique_ptr<SQLite3_result> resultset { get_admin_runtime_mysql_servers(mydb) };
				string mysrvs_checksum { get_checksum_from_hash(resultset ? resultset->raw_checksum() : 0) };
				save_runtime_mysql_servers(resultset.release());

				// PROPOSAL: Required to update the runtime_mysql_servers checksum with the new checksum, in a
				// similar way it's done during `commit` via `commit_update_checksum_from_mysql_servers`
	    	    uint64_t raw_checksum = this->runtime_mysql_servers ? this->runtime_mysql_servers->raw_checksum() : 0;
	    	    table_resultset_checksum[HGM_TABLES::MYSQL_SERVERS] = raw_checksum;

				// PROPOSAL: Required to update with the new 'mysql_group_replication_hostgroups'; this is
				// only required for preserving coherence in the checksums, otherwise they would be
				// inconsistent with `commit` generated checksums. This should be reworked into a function.
				{
					SpookyHash rep_hgs_hash {};
					bool init = false;
					uint64_t servers_v2_hash = table_resultset_checksum[HGM_TABLES::MYSQL_SERVERS_V2];

					if (servers_v2_hash) {
						if (init == false) {
							init = true;
							rep_hgs_hash.Init(19, 3);
						}
				
						rep_hgs_hash.Update(&servers_v2_hash, sizeof(servers_v2_hash));
					}

					CUCFT1(
						rep_hgs_hash, init, "mysql_replication_hostgroups", "writer_hostgroup",
						table_resultset_checksum[HGM_TABLES::MYSQL_REPLICATION_HOSTGROUPS]
					);
				}

				proxy_info("Checksum for table %s is %s\n", "mysql_servers", mysrvs_checksum.c_str());

				pthread_mutex_lock(&GloVars.checksum_mutex);
				update_glovars_mysql_servers_checksum(mysrvs_checksum);
				pthread_mutex_unlock(&GloVars.checksum_mutex);
			}
```

The comments with `PROPOSAL` mark the proposed changes to the previously mentioned required block. All this code is a proposal, and has not being tested.